OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -58,6 +60,27 @@ public void WorldLoaded(World w, WorldRenderer wr)
 
 			foreach (var cell in Map.AllCells)
 				UpdateCell(cell);
+
+			// Build the sprite layer dictionary for rendering resources
+			// All resources that have the same palette must also share a sheet and blend mode
```

Comment:
`SmudgeLayer` is horrible, and needs to be thrown out and replaced, so a similar comment applies there too.